lib/watcher.js

```diff
@@ -11,20 +11,15 @@ const log = require('./logger').create('watcher')
 const DIR_SEP = require('path').sep
 
 function watchPatterns (patterns, watcher) {
-  let pathsToWatch = new Set()
-
-  // expand ['a/{b,c}'] to ['a/b', 'a/c']
-  expandBraces(patterns)
-    .forEach((pattern) => pathsToWatch.add(PatternUtils.getBaseDir(pattern)))
-
-  pathsToWatch = Array.from(pathsToWatch)
-  // watch only common parents, no sub paths
-  pathsToWatch.forEach((path) => {
-    if (!pathsToWatch.some((p) => p !== path && path.startsWith(p + DIR_SEP))) {
-      watcher.add(path)
-      log.debug(`Watching "${path}"`)
-    }
-  })
+  expandBraces(patterns) // expand ['a/{b,c}'] to ['a/b', 'a/c']
+    .map(PatternUtils.getBaseDir)
+    .filter((value, index, array) => array.indexOf(value) === index) // filter unique values
```

Comment:
`value` -> `path` and `array` -> `paths` would make `paths.indexOf(path) === index` clearer.

Reply:
Done